client/src/components/App/App.tsx

```diff
@@ -27,30 +31,86 @@ const userNameClaimType = 'name';
 
 const App: React.FC = (): JSX.Element => {
 
+    const user = useSelector<StoreStateType, User>(state => state.user);
+
+    const [isUserUpdated, setIsUserUpdated] = useState<boolean>(true);
+
+    const setCurrentAndGroupUsers = (userId: string, userName: string, userToken: string) => {
+        axios.get(`/users/userById`).then((result: any) => {
+
+            const user = result && result.data && result.data.userById;
+            setUser({
+                ...user,
+                id: userId,
+                userName: userName,
+                token: userToken,
+            });
+
+            setIsUserUpdated(true);
+            return user;
+
+        }).then((user: any) => {
+            user && user.isAdmin && axios.get(`/users/getAllGroupUsers?investigationGroup=${user.investigationGroup}`)
```

Comment:
does it have only one investigationGroup? 
if so, consider adding it to the response.locals.group / response.locals.user.group in the admin middleware

Reply:
i mean - you already have user.investigationGroup in the server, so you dont need to send it as param

Reply:
unless there are several investigation Groups that you would like to get _separately_